### PR TITLE
persistent scene setup

### PR DESCRIPTION
**Pull request by treevg**

### CODE REVIEW

File: Assets/2D User Interface/Scripts/LogoAnimator.cs [hunk 1]
```diff
@@ -17,7 +18,8 @@ public class LogoAnimator : MonoBehaviour
         [SerializeField] private AnimationCurve FadeInCurve = AnimationCurve.EaseInOut(0f, 0f, 0.5f, 1f);
         [SerializeField] private AnimationCurve FadeOutCurve = AnimationCurve.EaseInOut(0f, 1f, 0.5f, 0f);
 
-        [SerializeField] public UnityEvent OnSplashScreensShown;
+        [SerializeField] private SceneTransitionController sceneTransitionController;
```
Comment (codewing) [inline Assets/2D User Interface/Scripts/LogoAnimator.cs:21]: Why did you change this? Can't you do exactly the same with the event? You can set the event to call the SceneManager.LoadScene method. In there you can set the name of the next level. The SceneManager should handle the fadeout and then load the next level. everything is already in there and its not as tightly coupled as this new solution?

Comment (treevg) [reply Assets/2D User Interface/Scripts/LogoAnimator.cs:21]: My plan was to use a SceneAsset  instead of an 'unmaintained' string to identify the scene. You couldn't link the scene object to the event though. That's why i changed it to a mere SceneAsset attribute. Later as i realized it can't be built  because SceneAssets only exist in the UnityEditor scope i reverted back to string.

Comment (treevg) [reply Assets/2D User Interface/Scripts/LogoAnimator.cs:21]: But now looking at it, you should mind the whole context: The LogoAnimator is very specific. It's there for one thing: Show logos, than change scene, so there is no issue with having a specific implementation either. The flexible event has now been moved one generalized instance further, to the SceneTransitionController which used to be the tightly coupled one in the previous implementation.

Comment (treevg) [reply Assets/2D User Interface/Scripts/LogoAnimator.cs]: To clarify i renamed some classes: 
LogoAnimator -> SplashScreenBehavior
SceneTransitionController -> FadeTransitionController

**Review by codewing — APPROVED** [review]
